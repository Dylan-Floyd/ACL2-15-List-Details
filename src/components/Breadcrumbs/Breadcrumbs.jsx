import React from 'react'
import { Link, useLocation } from 'react-router-dom'

export default function Breadcrumbs() {
  const location = useLocation()
  const dirs = location.pathname.slice(1).split('/')

  return (
    <h4>
      <Link to='/'>Home</Link>
      {dirs.map((dir, i) => (
        <span key={i}>
          {' / '}
          <Link to={'/'+dirs.slice(0, i+1).join('/')}>{dir}</Link>
        </span> 
      ))}
    </h4>
  )
}
