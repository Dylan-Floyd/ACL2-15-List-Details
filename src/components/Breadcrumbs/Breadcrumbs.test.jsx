import { render, screen } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import Breadcrumbs from './Breadcrumbs.jsx'

it('should render multiple paths correctly', async () => {
  const { container1 } = render(
    <MemoryRouter initialEntries={['/']}>
      <Breadcrumbs/>
    </MemoryRouter>
  )
  await screen.findByText(/Home/)
  expect(container1).toMatchSnapshot()

  const { container2 } = render(
    <MemoryRouter initialEntries={['/characters/2']}>
      <Breadcrumbs/>
    </MemoryRouter>
  )
  await screen.findByText(/characters/)
  await screen.findByText(/2/)
  expect(container2).toMatchSnapshot()
})