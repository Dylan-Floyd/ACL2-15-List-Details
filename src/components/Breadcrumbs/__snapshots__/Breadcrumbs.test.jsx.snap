// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render multiple paths correctly 1`] = `undefined`;

exports[`should render multiple paths correctly 2`] = `undefined`;
